File: digits_multiplication.py
# You are given a positive integer.
# Your function should calculate the product of the digits excluding any zeroes.
from functools import reduce


def checkio(n):
    return reduce((lambda x,y: x * y), [ int(c) for c in str(n) if c != '0' ])

print(checkio(123405) == 120)
print(checkio(999) == 729)
print(checkio(1000) == 1)
print(checkio(1111) == 1)

File: digits_multiplication.exs
# You are given a positive integer.
# Your function should calculate the product of the digits excluding any zeroes.
defmodule DigitsMultiplication do
  def checkio(n) do
    n
    |> Integer.to_string
    |> String.codepoints
    |> Stream.filter(&(&1 != "0"))
    |> Stream.map(&String.to_integer/1)
    |> Enum.into([])
    |> Enum.reduce(&*/2)
  end
end

IO.puts inspect DigitsMultiplication.checkio(123405) == 120
IO.puts inspect DigitsMultiplication.checkio(999) == 729
IO.puts inspect DigitsMultiplication.checkio(1000) == 1
IO.puts inspect DigitsMultiplication.checkio(1111) == 1
